File: android/app/src/main/java/com/anonymous/WalletApp/DeviceLocaleModule.kt
package com.anonymous.WalletApp

import com.facebook.react.bridge.Promise
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod
import java.util.*

class DeviceLocaleModule(reactContext: ReactApplicationContext) :
        ReactContextBaseJavaModule(reactContext) {

    override fun getName(): String {
        return "DeviceLocale"
    }

    @ReactMethod
    fun getDeviceLocale(promise: Promise) {
        try {
            val locale: Locale = Locale.getDefault()
            val language = locale.language // Ex: "en"
            val country = locale.country // Ex: "US"
            val localeString = "$language-$country"
            promise.resolve(localeString)
        } catch (e: Exception) {
            promise.reject("Error", e)
        }
    }
}

File: ios/DeviceLocaleModule.swift
//
//  DeviceLocaleModule.swift
//  WalletApp
//
//  Created by julio on 09/11/24.
//

import Foundation
import React

@objc(DeviceLocale)
class DeviceLocale: NSObject {
  
  // Método exposto ao React Native para obter o idioma e região do dispositivo
  @objc
  func getDeviceLocale(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
    let locale = Locale.current
    let language = locale.languageCode ?? "en" // Padrão para "en" se não estiver disponível
    let country = locale.regionCode ?? "US"    // Padrão para "US" se não estiver disponível
    let localeString = "\(language)-\(country)"
    resolve(localeString)
  }
  
  @objc
  static func requiresMainQueueSetup() -> Bool {
    return false
  }
}
